<a href="https://colab.research.google.com/github/evyrinnnx7/DataPenjualanKantinSekolah/blob/main/Data_Penjualan_Kantin_Sekolah.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
# LATIHAN 1: Operasi Dasar NumPy Array
import numpy as np

harga = np.array([5000, 7000, 3000, 12000, 4500])
print('Rata-rata harga:', harga.mean())
print('Harga tertinggi:', harga.max())
print('Harga setelah diskon 10%:', harga * 0.9)


Rata-rata harga: 6300.0
Harga tertinggi: 12000
Harga setelah diskon 10%: [ 4500.  6300.  2700. 10800.  4050.]


**Analisis Latihan 1**

Perintah harga * 0.9 pada NumPy Array langsung mengalikan seluruh angka di dalamnya sekaligus. Pada List Python biasa, kodenya akan error karena list tidak bisa dikalikan dengan angka desimal.

In [2]:
# LATIHAN 2: Membuat Series & DataFrame
import pandas as pd

data_kantin = {
    'menu': ['Nasi Goreng', 'Es Teh', 'Mie Ayam', 'Es Teh', None],
    'harga': [12000, 4000, 10000, 4000, 8000],
    'terjual': [23, 40, None, 35, 18]
}

df_latihan2 = pd.DataFrame(data_kantin)
print(df_latihan2)

          menu  harga  terjual
0  Nasi Goreng  12000     23.0
1       Es Teh   4000     40.0
2     Mie Ayam  10000      NaN
3       Es Teh   4000     35.0
4         None   8000     18.0


**Analisis Latihan 2**

Data kosong berada di kolom menu (baris ke-4) dan terjual (baris ke-2). Jika tidak dibersihkan, hasil perhitungan statistik seperti rata-rata dan total pendapatan akan menjadi salah.


In [7]:
# LATIHAN 3: Data Loading & Inspection
df = pd.read_csv('data_kantin - data_kantin.csv')

print(df.head())
print(df.info())
print(df.describe())
print(df.shape)

      tanggal         menu kategori  harga  terjual
0  2026-08-20   Teh Hangat  Minuman   3000     11.0
1  2026-08-31  Nasi Goreng  Makanan  12000     29.0
2  2026-08-21  Nasi Goreng  Makanan  12000     28.0
3  2026-08-28   Teh Hangat  Minuman   3000     12.0
4  2026-08-21     Es Jeruk  Minuman   5000     19.0
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 110 entries, 0 to 109
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   tanggal   110 non-null    object 
 1   menu      107 non-null    object 
 2   kategori  110 non-null    object 
 3   harga     110 non-null    int64  
 4   terjual   104 non-null    float64
dtypes: float64(1), int64(1), object(3)
memory usage: 4.4+ KB
None
              harga     terjual
count    110.000000  104.000000
mean    6336.363636   22.923077
std     3520.001232   10.831178
min     2000.000000    8.000000
25%     3000.000000   13.000000
50%     5000.000000   20.000000
75%    10000.000000 

**Analisis Latihan 3**

Kolom menu dan terjual punya jumlah data terisi yang lebih sedikit dari total 110 baris. Ini menandakan ada data kosong (missing value) pada kedua kolom tersebut.

In [ ]:
# LATIHAN 4: Menangani Missing Value

print(df.isnull().sum())
df['terjual'] = df['terjual'].fillna(0)
df = df.dropna(subset=['menu'])


**Analisis Latihan 4**

Kolom terjual diisi angka 0 (fillna) karena diasumsikan tidak ada transaksi penjualan. Baris dengan menu kosong dihapus (dropna) agar tidak menebak nama produk yang membuat laporan tidak akurat.

In [9]:
# LATIHAN 5: Menangani Duplikat dan Tipe Data

print(df.duplicated().sum())
df = df.drop_duplicates()

df['harga'] = df['harga'].astype(int)
print(df.dtypes)

0
tanggal      object
menu         object
kategori     object
harga         int64
terjual     float64
dtype: object


**Analisis Latihan 5**

Jumlah data berkurang dari 110 menjadi 102 baris setelah duplikat dihapus. Memastikan tipe data benar sangat penting agar perhitungan angka keuangan berjalan tepat tanpa error.

In [10]:
# LATIHAN 6: Data Manipulation (Filtering, Sorting, Groupby)
laris = df[df['terjual'] > 20]
urut = df.sort_values(by='terjual', ascending=False)

df['total_pendapatan'] = df['harga'] * df['terjual']

ringkasan = df.groupby('menu')['total_pendapatan'].sum() # agregasi
print(ringkasan)


menu
Es Jeruk       1460000.0
Es Teh         2144000.0
Kerupuk         972000.0
Mie Ayam       2410000.0
Nasi Goreng    3924000.0
Roti Bakar     1288000.0
Teh Hangat      459000.0
Name: total_pendapatan, dtype: float64


**Analisis Latihan 6**

Menu dengan pendapatan tertinggi adalah Nasi Goreng dengan total omzet Rp 3.924.000. Informasi ini berguna untuk menjaga stok bahan baku Nasi Goreng dan membuat strategi paket promo bundling.